packages/replayio/src/utils/createDeferred.ts

```diff
@@ -6,8 +6,8 @@ export type StatusPending = typeof STATUS_PENDING;
 export type StatusRejected = typeof STATUS_REJECTED;
 export type StatusResolved = typeof STATUS_RESOLVED;
 
-export interface Deferred<Type, Data = void> {
```

Comment:
`void` is somewhat a weird type in TS and honestly I don't even understand it in full despite being deep into their internals. My rule of thumb is to avoid it and use `unknown`/`undefined` (depending on the context) - unless absolutely necessary to use `void` (it's sometimes handy to use it as return type of callbacks, so there is that)